src/jit/codegencommon.cpp

```diff
@@ -3811,20 +3811,29 @@ void            CodeGen::genFnPrologCalleeRegArgs(regNumber xtraReg,
         // So, for that case we retain the type of the register in the regArgTab.
         // In other cases, we simply use the type of the lclVar to determine the type of the register.
 
+#ifdef FEATURE_UNIX_AMD64_STRUCT_PASSING
+
+        // This is the UNIX_AMD64 implementation
         var_types   getRegType(Compiler* compiler)
         {
-#if defined(FEATURE_UNIX_AMD64_STRUCT_PASSING)
-            return type;
-#elif defined(_TARGET_ARM_)
-            LclVarDsc varDsc = compiler->lvaTable[varNum];
-            return varDsc.lvIsHfaRegArg ? varDsc.GetHfaType() : varDsc.lvType;
+            return type;  // UNIX_AMD64 implementation
+        }
 
-    // TODO-ARM64: Do we need the above to handle HFA structs on ARM64?
+#else // !FEATURE_UNIX_AMD64_STRUCT_PASSING
 
-#else // !_TARGET_ARM_
-            return compiler->lvaTable[varNum].lvType;
-#endif // !_TARGET_ARM_
+        // This is the implementation for all other targets
+        var_types   getRegType(Compiler* compiler)
+        {
+            LclVarDsc varDsc = compiler->lvaTable[varNum];
+#ifdef FEATURE_HFA
```

Comment:
This #ifdef could be removed since now we have an accessor to lvIsHfaRegArg.  Plus we can combine implementation of unix and arm like before?